packages/components/src/components/as-category-widget/as-category-widget.tsx

```diff
@@ -0,0 +1,203 @@
+import { Component, Event, EventEmitter, Method, Prop, State } from '@stencil/core';
+import readableNumber from '../../utils/readable-number';
+import { shadeOrBlend } from '../../utils/styles';
+
+const OTHER_CATEGORY_COLOR = '#747474';
+const OTHER_CATEGORY_NAME = 'Other';
+const DEFAULT_BAR_COLOR = '#47DB99';
+
+@Component({
+  shadow: false,
+  styleUrl: './as-category-widget.scss',
+  tag: 'as-category-widget',
+})
+export class CategoryWidget {
+  @Prop() public categories: object[] = [];
+  @Prop() public defaultBarColor: string = DEFAULT_BAR_COLOR;
+  @Prop() public description: string;
+  @Prop() public heading: string;
+  @Prop() public showClearButton: boolean;
+  @Prop() public showHeader: boolean = true;
+  @Prop() public useTotalPercentage: boolean = false;
+
+  @Event() public categoriesSelected: EventEmitter<string[]>;
+
+  @State() private selectedCategories: string[] = [];
+  @State() private numberOfVisibleCategories: number = 5;
+
+  @Method()
+  public getSelectedCategories() {
+    return this.selectedCategories;
+  }
+
+  @Method()
+  public clearSelection() {
+    if (!this.selectedCategories.length) {
+      return;
+    }
+
+    this.selectedCategories = [];
+    this._onCategoriesChanged();
+  }
+
+  public render() {
+    return [
+      this._renderHeader(),
+      <ul class='as-category-widget__list'>{this._renderCategories()}</ul>,
```

Comment:
Is this too much? I'd love to have 3 equal methods 
```jsx
private function _renderContent() {
  return <ul class='as-category-widget__list'>{this._renderCategories()}</ul>
}
```

Reply:
No, I thought about it too, would be good :)

I feel like I did a poor function ordering in the class, I think it is convoluted somehow 🤷‍♂️